### PR TITLE
dep: update com_github_wasmtime: v0.32.0 (current: v0.31.0)

### LINKED ISSUE #19279: Newer release available `com_github_wasmtime`: v0.32.0 (current: v0.31.0)
Package Name: com_github_wasmtime
Current Version: v0.31.0@2021-10-29 14:34:56
Available Version: v0.32.0@2021-12-13 19:47:30
Upstream releases: https://github.com/bytecodealliance/wasmtime/releases

### PR DESCRIPTION
**Pull request by kfaseela**
Signed-off-by: Faseela K <faseela.k@est.tech>

Fixes #19279

### CODE REVIEW

**Comment by moderation** [comment]
/lgtm deps

Dependency updates look OK but CI failures look real. You might need some of the WASM experts to chime in on what changed between releases - https://github.com/bytecodealliance/wasmtime/blob/main/RELEASES.md#0320

**Comment by phlax** [comment]
cc @PiotrSikora

**Comment by PiotrSikora** [comment]
Wasmtime update requires changes in Proxy-Wasm, please at the previous updates:
- v0.30: https://github.com/proxy-wasm/proxy-wasm-cpp-host/pull/194, https://github.com/envoyproxy/envoy/pull/18499
- v0.31: https://github.com/proxy-wasm/proxy-wasm-cpp-host/pull/196, https://github.com/envoyproxy/envoy/pull/18860

**Comment by kfaseela** [comment]
> Wasmtime update requires changes in Proxy-Wasm, please look at the previous updates:
> 
> * v0.30: [wasmtime: update to v0.30.0. proxy-wasm/proxy-wasm-cpp-host#194](https://github.com/proxy-wasm/proxy-wasm-cpp-host/pull/194), [deps: Bump `com_github_wasmtime` -> 0.30.0 #18499](https://github.com/envoyproxy/envoy/pull/18499)
> * v0.31: [wasmtime: update to v0.31.0. proxy-wasm/proxy-wasm-cpp-host#196](https://github.com/proxy-wasm/proxy-wasm-cpp-host/pull/196), [wasm: update Wasmtime to v0.31.0. #18860](https://github.com/envoyproxy/envoy/pull/18860)

Thanks for the info @PiotrSikora . Will wait for the proxy-wasm update then. I couldn't see a corresponding commit for 0.32.0 there.

**Comment by PiotrSikora** [comment]
@kfaseela you pretty much need to do the update yourself, but it's pretty easy (sans breaking changes) and there are instructions in the PRs referenced above.

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -1027,8 +1027,8 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "WebAssembly for Proxies (C++ host implementation)",
         project_desc = "WebAssembly for Proxies (C++ host implementation)",
         project_url = "https://github.com/proxy-wasm/proxy-wasm-cpp-host",
-        version = "f38347360feaaf5b2a733f219c4d8c9660d626f0",
-        sha256 = "bf10de946eb5785813895c2bf16504afc0cd590b9655d9ee52fb1074d0825ea3",
+        version = "ab742cd64f9839e70dd194581e193dadf158928a",
+        sha256 = "0f721d7fd897ed904acfbcfccd43d1a79567084d8234c210e11c370226b0d702",
```
Comment (PiotrSikora) [inline bazel/repository_locations.bzl:1031]: Could you update this to the merged commit? Thanks!

Comment (kfaseela) [reply bazel/repository_locations.bzl:1031]: @PiotrSikora : Do I have to wait for [#209](https://github.com/proxy-wasm/proxy-wasm-cpp-host/pull/209) to be merged?

Comment (PiotrSikora) [reply bazel/repository_locations.bzl:1031]: No, it doesn't matter, since we're updating the tarball that's updated in #209 in this PR.

**Review by PiotrSikora — APPROVED** [review]
Thanks!

**Review by lizan — APPROVED** [review]

**Pull request merged**